Mathlib/Analysis/NormedSpace/Extend.lean

```diff
@@ -0,0 +1,181 @@
+      rw [← smul_eq_mul, ← map_smul, extendTo𝕜'_apply_re]
+#align linear_map.norm_extend_to_𝕜'_apply_sq LinearMap.norm_extendTo𝕜'_apply_sq
+
+end LinearMap
+
+namespace ContinuousLinearMap
+
+variable [NormedSpace ℝ F] [IsScalarTower ℝ 𝕜 F]
+
+/-- The norm of the extension is bounded by `‖fr‖`. -/
+theorem norm_extendTo𝕜'_bound (fr : F →L[ℝ] ℝ) (x : F) :
+    ‖(fr.toLinearMap.extendTo𝕜' x : 𝕜)‖ ≤ ‖fr‖ * ‖x‖ := by
+  set lm : F →ₗ[𝕜] 𝕜 := fr.toLinearMap.extendTo𝕜'
+  by_cases h : lm x = 0
+  · rw [h, norm_zero]
+    apply mul_nonneg <;> exact norm_nonneg _
+  rw [← mul_le_mul_left (norm_pos_iff.2 h), ← sq]
+  calc
+    ‖lm x‖ ^ 2 = fr (conj (lm x : 𝕜) • x) := fr.toLinearMap.norm_extendTo𝕜'_apply_sq x
+    _ ≤ ‖fr (conj (lm x : 𝕜) • x)‖ := (le_abs_self _)
+    _ ≤ ‖fr‖ * ‖conj (lm x : 𝕜) • x‖ := (le_op_norm _ _)
+    _ = ‖(lm x : 𝕜)‖ * (‖fr‖ * ‖x‖) := by rw [norm_smul, norm_conj, mul_left_comm]
+#align continuous_linear_map.norm_extend_to_𝕜'_bound ContinuousLinearMap.norm_extendTo𝕜'_bound
+
+/-- Extend `fr : F →L[ℝ] ℝ` to `F →L[𝕜] 𝕜`. -/
+noncomputable def extendTo𝕜' (fr : F →L[ℝ] ℝ) : F →L[𝕜] 𝕜 :=
+  LinearMap.mkContinuous _ ‖fr‖ fr.norm_extendTo𝕜'_bound
+#align continuous_linear_map.extend_to_𝕜' ContinuousLinearMap.extendTo𝕜'
+
+theorem extendTo𝕜'_apply (fr : F →L[ℝ] ℝ) (x : F) :
+    fr.extendTo𝕜' x = (fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜) := rfl
+#align continuous_linear_map.extend_to_𝕜'_apply ContinuousLinearMap.extendTo𝕜'_apply
+
+@[simp]
+theorem norm_extendTo𝕜' (fr : F →L[ℝ] ℝ) : ‖(fr.extendTo𝕜' : F →L[𝕜] 𝕜)‖ = ‖fr‖ :=
+  le_antisymm (LinearMap.mkContinuous_norm_le _ (norm_nonneg _) _) <|
+    op_norm_le_bound _ (norm_nonneg _) fun x =>
+      calc
+        ‖fr x‖ = ‖re (fr.extendTo𝕜' x : 𝕜)‖ := congr_arg norm (fr.extendTo𝕜'_apply_re x).symm
+        _ ≤ ‖(fr.extendTo𝕜' x : 𝕜)‖ := (abs_re_le_norm _)
+        _ ≤ ‖(fr.extendTo𝕜' : F →L[𝕜] 𝕜)‖ * ‖x‖ := le_op_norm _ _
+#align continuous_linear_map.norm_extend_to_𝕜' ContinuousLinearMap.norm_extendTo𝕜'
+
+end ContinuousLinearMap
+
+/-- Extend `fr : RestrictScalars ℝ 𝕜 F →ₗ[ℝ] ℝ` to `F →ₗ[𝕜] 𝕜`. -/
+noncomputable def LinearMap.extendTo𝕜 (fr : RestrictScalars ℝ 𝕜 F →ₗ[ℝ] ℝ) : F →ₗ[𝕜] 𝕜 :=
+  -- Porting note: Added instance for `NormedSpace 𝕜 (RestrictScalars ℝ 𝕜 F)`
+  let _ : NormedSpace 𝕜 (RestrictScalars ℝ 𝕜 F) := by
+    unfold RestrictScalars
+    infer_instance
+  fr.extendTo𝕜'
+#align linear_map.extend_to_𝕜 LinearMap.extendTo𝕜
+
+theorem LinearMap.extendTo𝕜_apply (fr : RestrictScalars ℝ 𝕜 F →ₗ[ℝ] ℝ) (x : F) :
+    fr.extendTo𝕜 x = (fr x : 𝕜) - (I : 𝕜) * (fr ((I : 𝕜) • x) : 𝕜) := rfl
+#align linear_map.extend_to_𝕜_apply LinearMap.extendTo𝕜_apply
+
+/-- Extend `fr : RestrictScalars ℝ 𝕜 F →L[ℝ] ℝ` to `F →L[𝕜] 𝕜`. -/
+noncomputable def ContinuousLinearMap.extendTo𝕜 (fr : RestrictScalars ℝ 𝕜 F →L[ℝ] ℝ) : F →L[𝕜] 𝕜 :=
+  -- Porting note: Added instance for `NormedSpace 𝕜 (RestrictScalars ℝ 𝕜 F)`
+  let _ : NormedSpace 𝕜 (RestrictScalars ℝ 𝕜 F) := by
```

Comment:
If this isn't synthesized, shouldn't it be added as a global instance?

Reply:
Done